### PR TITLE
fix(vscode): Use same ports and child proccess in designer and data mapper extension

### PR DESCRIPTION
**Pull request by ccastrotrejo**
This pull request to `vs-code-designer` includes various changes to improve the codebase, including refactoring, renaming constants, and updating functions to use new variables. The most important changes include updating the `waitForBackendRuntimeStartUp` and `startBackendRuntime` functions to use new variables, and refactoring the `openDesignerForLocalProject` workflow command to use a single `designTimePort` variable.

Main interface changes:

* `apps/vs-code-designer/src/app/commands/dataMapper/FxWorkflowRuntime.ts`: Updated `waitForBackendRuntimeStartUp` and `startBackendRuntime` functions to use new variables. (Fbd868c6)
* <a href="diffhunk://#diff-fc7c0af4bc38480e2168d96396fd63202ae1366ace724fe82d13c8fa99125a26L291-R291">`apps/vs-code-designer/src/app/commands/workflows/openDesigner/openDesignerForLocalProject.ts`</a>: Refactored the `openDesignerForLocalProject` workflow command to use a single `designTimePort` variable. <a href="diffhunk://#diff-fc7c0af4bc38480e2168d96396fd63202ae1366ace724fe82d13c8fa99125a26L291-R291">[1]</a> <a href="diffhunk://#diff-fc7c0af4bc38480e2168d96396fd63202ae1366ace724fe82d13c8fa99125a26L93-R93">[2]</a>

Configuration improvements:

* <a href="diffhunk://#diff-9022c985134376e8dc5f40c6ae9f15c05aa17d3f7a8b8a8e08dc89d7e99bfed3L23-L28">`apps/vs-code-designer/src/extensionVariables.ts`</a>: Renamed a constant from `workflowDesignTimeDir` to `designTimeDirectoryName`.
* <a href="diffhunk://#diff-7a497a1a7a0737f81ae4b301851aee2130c51754079e63805b8d936e6b3508f3L24-R24">`apps/vs-code-designer/src/constants.ts`</a>: Renamed constant `workflowDesignTimeDir` to `designTimeDirectoryName`.
* `apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createLogicApp/workflowDesigner/DesignerConfig.ts`: Refactored `DesignerConfig.ts` to use `designTimeDirectoryName` constant instead of hardcoding directory name in multiple places. (F2c31844)

Other changes:

* `apps/vs-code-designer/src/app/utils/codeless/startDesignTimeApi.ts`: Refactored `startDesignTimeApi.ts` to use `designChildProcess` instead of `workflowDesignChildProcess`, exported functions, and made other minor changes. (F62a568c)
* <a href="diffhunk://#diff-0aebfb1224bb53a0c28b7edabe0664ca774957126a689d88c183a4142cd43d92L96-R96">`apps/vs-code-designer/src/app/commands/dataMapper/DataMapperPanel.ts`</a>: Updated `DataMapperPanel` function to use `designTimePort` instead of `dataMapperRuntimePort`.
* `apps/vs-code-designer/src/main.ts`: Removed the `stopDataMapperBackend` function from `main.ts` and its call from the `deactivate` function. (F0f63e2d)
* <a href="diffhunk://#diff-d337a3f857f729961bb5cad56c4ff6f0ed0cfd50d4b213df062c179a1c178b13R5">`apps/vs-code-designer/src/app/utils/vsCodeConfig/tasks.ts`</a>: Includes changes to the `tasks.ts` file in the `vs-code-designer` app, including removing an import, adding error handling, and appending an error message to the output channel. <a href="diffhunk://#diff-d337a3f857f729961bb5cad56c4ff6f0ed0cfd50d4b213df062c179a1c178b13R5">[1]</a> <a href="diffhunk://#diff-d337a3f857f729961bb5cad56c4ff6f0ed0cfd50d4b213df062c179a1c178b13R90-R91">[2]</a> <a href="diffhunk://#diff-d337a3f857f729961bb5cad56c4ff6f0ed0cfd50d4b213df062c179a1c178b13R123-R126">[3]</a>

### CODE REVIEW

**Review by Elaina-Lee — APPROVED** [review]

File: apps/vs-code-designer/src/app/commands/dataMapper/FxWorkflowRuntime.ts [hunk 1]
```diff
@@ -89,60 +81,39 @@ async function createJsonFile(
 }
 
 async function waitForBackendRuntimeStartUp(url: string, initialTime: number): Promise<void> {
```
Comment (Elaina-Lee) [inline apps/vs-code-designer/src/app/commands/dataMapper/FxWorkflowRuntime.ts:83]: This one now looks practically the same (except for a variable) to `waitForDesignTimeStartUp` in `startDesignTimeApi.ts`, might be worth coming back to it later to reuse the function!

File: apps/vs-code-designer/src/app/commands/dataMapper/FxWorkflowRuntime.ts [hunk 2]
```diff
@@ -89,60 +81,39 @@ async function createJsonFile(
...
-async function isBackendRuntimeUp(url: string): Promise<boolean> {
-  try {
-    await fetch(url);
-    return Promise.resolve(true);
-  } catch (ex) {
-    return Promise.resolve(false);
-  }
-}
-
 function startBackendRuntimeProcess(workingDirectory: string | undefined, command: string, ...args: string[]): void {
```
Comment (Elaina-Lee) [inline apps/vs-code-designer/src/app/commands/dataMapper/FxWorkflowRuntime.ts:95]: This one seems to contain quite a bit of duplicates to `startDesignTimeProcess` in `startDesignTimeApi.ts`, might be worth coming back to it later to reuse the function!

File: apps/vs-code-designer/src/app/commands/dataMapper/FxWorkflowRuntime.ts (hunk 1 above)
Comment (ccastrotrejo) [reply apps/vs-code-designer/src/app/commands/dataMapper/FxWorkflowRuntime.ts:83]: Yees! I will update the logic to reuse the `waitForDesignTimeStartUp` function. Thanks!

File: apps/vs-code-designer/src/app/commands/dataMapper/FxWorkflowRuntime.ts (hunk 2 above)
Comment (ccastrotrejo) [reply apps/vs-code-designer/src/app/commands/dataMapper/FxWorkflowRuntime.ts:95]: True, I will update the code to reuse it. Thanks!

**Pull request merged**